### DISCUSSION TITLE
How do I make the scroll of a table start at the end?

### DISCUSSION THREAD

**Discussion by bridgecode** [element-plus/element-plus#16281] [Q&A] [answered]
Hello currently I have this
![image](https://github.com/element-plus/element-plus/assets/7904191/d4ef2672-9377-431f-b3ca-6e21a22b2345)

I would LIKE the table (aside from the sticky 'Habit' column) to start at the end so the user scrolls backward to get to past days, like this 
![image](https://github.com/element-plus/element-plus/assets/7904191/dc9d57d8-c3b2-4152-aa5a-ccc9c4f544d7)

Is this possible, I'm having no luck with the current options

Here's my code:
![image](https://github.com/element-plus/element-plus/assets/7904191/82b912ea-7e2d-4d39-9bd0-139a4157968d)

**Comment by Aseisman** [comment, marked as answer]
https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXRhYmxlIDpkYXRhPVwidGFibGVEYXRhXCIgc3R5bGU9XCJ3aWR0aDogMTAwJVwiIGhlaWdodD1cIjI1MFwiIHJlZj1cInRhYmxlUmVmXCI+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBmaXhlZCBwcm9wPVwiZGF0ZVwiIGxhYmVsPVwiRGF0ZVwiIHdpZHRoPVwiMTUwXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJuYW1lXCIgbGFiZWw9XCJOYW1lXCIgd2lkdGg9XCIxMjBcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cInN0YXRlXCIgbGFiZWw9XCJTdGF0ZVwiIHdpZHRoPVwiMTIwXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJjaXR5XCIgbGFiZWw9XCJDaXR5XCIgd2lkdGg9XCIzMjBcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImFkZHJlc3NcIiBsYWJlbD1cIkFkZHJlc3NcIiB3aWR0aD1cIjYwMFwiIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiemlwXCIgbGFiZWw9XCJaaXBcIiB3aWR0aD1cIjEyMFwiIC8+XG4gIDwvZWwtdGFibGU+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSBcInZ1ZVwiO1xuXG5jb25zdCB0YWJsZVJlZiA9IHJlZigpO1xuc2V0VGltZW91dCgoKT0+e1xuICB0YWJsZVJlZi52YWx1ZS5zZXRTY3JvbGxMZWZ0KHRhYmxlUmVmLnZhbHVlLiRlbC5jbGllbnRXaWR0aClcbn0sMjAwKVxuXG5cbmNvbnN0IHRhYmxlRGF0YSA9IFtcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAzJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDInLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDgnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTA3JyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319

**Reply by Aseisman** [reply]
see the demo

**Reply by bridgecode** [reply]
Just now seeing this, thank you so much!

**Reply by bridgecode** [reply]
This is the answer and is aligned with the other suggested fixes online, but for me does not work. I think something with the v-for OR nested elements breaks the reference to the nested element and this doesnt seem to work. Could be user error but I choose life and elected to use a basic table and row element inside of the ion-card and be done with the headache of the el-table component. 

Heres my solution for anyone that wants to just have it work and not bother with this.

` <ion-card ref="tableWrapper" id="bullet-container" class="table-scroll-container">

      <table id="bullet-container-main" class="table-main" style="width: 100%;">

        <thead>
          <tr>
            <th class="col-main" style="width: 120px; position: sticky; left: 0; background: white;">Habit</th>
            <th v-for="day in dayColumns" :key="day" class="col-date" style="width: 60px;">{{ day }}</th>
          </tr>
        </thead>
        <tbody>
          <tr v-for="(row, index) in TableData" :key="index">
            <td class="col-main" style="width: 120px; position: sticky; left: 0; background: white;">{{ row.date }}</td>
            <td v-for="day in dayColumns" :key="day" class="col-date" style="width: 60px;">
              <ion-icon class="habit-completed" v-if="Math.random() < .5" :icon="checkmarkCircle"></ion-icon>
              <ion-icon class="habit-missed" v-else :icon="remove"></ion-icon>
            </td>
          </tr>
        </tbody>
      </table>
    </ion-card>`


SCSS:
`$font-family: Arial, sans-serif;
$font-size: 1.2em;

$primary-color: var(--ion-color-primary);
$primary-contrast-color: var(--ion-color-primary-contrast);
$secondary-color: var(--ion-color-secondary);
$secondary-shade-color: var(--ion-color-secondary-shade);
$primary-text-color: var(--ion-color-primary-text);

$table-border-color: $secondary-shade-color;
$table-border: 1px solid $table-border-color;
$table-header-bg-color: #f4f4f4;
$table-body-bg-color-even: #f9f9f9;
$table-body-bg-color-odd: #fff;

ion-icon {
  font-size: $font-size;
}

.table-scroll-container {
  background-color: $primary-color;
  color: $primary-contrast-color;
  overflow: hidden;
  padding: 2px;
  overflow-x: auto;
  border: $table-border !important;
  margin: 10px 0px;
}

.table-main {
  color: $primary-text-color;
  font-family: poppins;
}

.col-main {
  background-color: $secondary-shade-color !important;
}

.col-date {
  background-color: $secondary-color !important;
}

.el-table__body,
.el-table__footer,
.el-table__header,
.el-scrollbar__wrap {
  background: $secondary-shade-color;
}

.el-table tr {
  background-color: var(--el-table-tr-bg-color);
  scroll-snap-align: right;
}

.el-table {
  --el-table-border-color: $table-border-color;
  --el-table-border: $table-border;
}

body {
  font-family: $font-family;
}

.table-main {
  width: 100%;
  border-collapse: collapse;
  table-layout: fixed;
}

.table-header-cell, .table-body-cell {
  border: $table-border;
  text-align: center;
  padding: 8px;
}

.table-header-cell {
  background-color: $table-header-bg-color;
}

.table-header-cell.col-main, .table-body-cell.col-main {
  position: sticky;
  left: 0;
  background-color: #fff;
  z-index: 2;
}

.table-header {
  z-index: 3;
}

.table-header-cell.col-main {
  z-index: 4;
}

.table-body .table-row:nth-child(even) .table-body-cell {
  background-color: $table-body-bg-color-even;
}

.table-body .table-row:nth-child(odd) .table-body-cell {
  background-color: $table-body-bg-color-odd;
}

.habit-completed {
    color: #00c310;
}

.habit-missed {
    color: black;
}`
